src/main/java/no/uio/ifi/localega/doa/rest/MetadataController.java

```diff
@@ -21,36 +27,32 @@
 public class MetadataController {
 
     @Autowired
-    private JWTVerifier jwtVerifier;
+    protected HttpServletRequest request;
 
     @Autowired
     private FileRepository fileRepository;
 
     @Autowired
     private DatasetRepository datasetRepository;
 
+    @SuppressWarnings("unchecked")
     @GetMapping("/datasets")
-    public ResponseEntity<?> datasets(@RequestHeader(value = "Authorization") String token) {
-        DecodedJWT decodedJWT = jwtVerifier.verify(token.replace("Bearer ", ""));
-        String subject = decodedJWT.getSubject();
-        log.info("User {} is authenticated and is getting the list of datasets", subject);
-        Set<String> datasetIds = Arrays.stream(decodedJWT.getClaim("authorities").asArray(String.class)).collect(Collectors.toSet());
+    public ResponseEntity<?> datasets() {
+        log.info("User has permissions to list datasets");
+        Set<String> datasetIds = (Set<String>) request.getAttribute(AAIAspect.DATASETS);
         Collection<LEGADataset> datasets = datasetRepository.findByDatasetIdIn(datasetIds);
         return ResponseEntity.ok(datasets.stream().map(LEGADataset::getDatasetId).collect(Collectors.toSet()));
     }
 
+    @SuppressWarnings("unchecked")
```

Comment:
curious what are the warnings that need suppressing ?

Reply:
It's suppressing Java warning about this cast: `Set<String> datasetIds = (Set<String>) request.getAttribute(AAIAspect.DATASETS);`. The request holds untyped object, but I have to cast it to the typed object - lack of API...